### PR TITLE
feat: optimize filters

### PR DESCRIPTION
**Pull request by devhaozi**
Closes # 

## 📑 Description

filters 支持嵌套结构体，同时将添加 filters 的操作挪到框架实现。

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -9,8 +9,8 @@ jobs:
     strategy:
       matrix:
         go:
-          - '1.21'
-          - '1.20'
+          - 'stable'
+          - 'oldstable'
```
Comment (hwbrzzl) [inline .github/workflows/test.yml:13]:
```suggestion
          - '1.21'
```

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -25,8 +25,8 @@ jobs:
...
```
Comment (hwbrzzl) [inline .github/workflows/test.yml:29]:
```suggestion
          - '1.21'
```

File: context_request.go
Comment (hwbrzzl) [inline context_request.go:1]: Please add some test cases for this change.

**Comment by devhaozi** [comment]
太久了，暂时关了，后面有需要再说。

**Pull request closed without merging**